### DISCUSSION TITLE
Задание 4

### DISCUSSION THREAD

**Discussion by eprilutskaya** [anastasiaprakapchuk/AmastyLearningNewTechnology#16] [General]
Amasty/AnastasiaModule/Controller/Autocomplete/Autocomplete.php:59
У коллекции можно устанавливать макс количество записей получаемое за раз `$productCollection->setPageSize(5) ` , таким образом коллекция разбивается на "страницы". Дефолтно установлено значение 20. 
Вам достаточно после применения фильтров выполнить setPageSize, в foreach вы будете получать только 5 записей.

Здесь есть пример работы с коллекцией с setPageSize
https://magento.stackexchange.com/questions/282649/magento-2-how-to-load-data-by-page-using-same-collection-object-again

Amasty/AnastasiaModule/view/frontend/web/js/koAutocomplete.js:34
processAutocomplete можно переделать на knockout, чтобы не получать элементы через jQuery:

К элементам в темплите через data-bind можно добавлять обработчки событий onClick.
К элементу списка можно привязать вашу функцию processAutocomplete, например:
`<li class="css-li-autocomplete" data-bind="click: $parent.processAutocomplete.bind($parent)">`
Где $parent - Componen
В функции при выполнении будет доступен this - Componen , тк выполнена привязка контекста .bind($parent)
В функцию будет приходить текущий knockout элемент, из которого можно получить sku или name и передать в searchText
```
processAutocomplete: function (element) {
    this.searchText(element.sku);
    this.isFocused(false);
},
```
Оценка 9

**Comment by anastasiaprakapchuk** [comment]
Спасибо большое за подробный и полезный комментарий.
Поправила Ваши замечания. 

В Amasty/AnastasiaModule/Controller/Autocomplete/Autocomplete.php:59 сделала таким образом:
```
$productCollection = $this->productCollectionFactory->create()
                ->addAttributeToSelect(['sku', 'name'])
                ->addAttributeToFilter('sku', ['like' => $sku . '%'])
                ->setPageSize(5);

            $products = [];
            foreach ($productCollection as $product) {
                $products[] = ['sku' => $product->getSku(), 'name' => $product->getName()];
            }
```
так, как `$products = $productCollection->getData();`  присылает не то, что хотелось бы, а нужны только 'sku' и 'name'.
